x-pack/plugins/maps/server/routes.js

```diff
@@ -52,25 +52,31 @@ const EMPTY_EMS_CLIENT = {
   addQueryParams() {},
 };
 
-export function initRoutes(router, licenseUid, emsSettings, kbnVersion, logger) {
+export function initRoutes(router, getLicenseId, emsSettings, kbnVersion, logger) {
   let emsClient;
-
-  if (emsSettings.isIncludeElasticMapsService()) {
-    emsClient = new EMSClient({
-      language: i18n.getLocale(),
-      appVersion: kbnVersion,
-      appName: EMS_APP_NAME,
-      fileApiUrl: emsSettings.getEMSFileApiUrl(),
-      tileApiUrl: emsSettings.getEMSTileApiUrl(),
-      landingPageUrl: emsSettings.getEMSLandingPageUrl(),
-      fetchFunction: fetch,
-    });
-    emsClient.addQueryParams({ license: licenseUid });
-  } else {
-    emsClient = EMPTY_EMS_CLIENT;
-  }
+  let lastLicenseId;
 
   function getEMSClient() {
+    const currentLicenseId = getLicenseId();
+    if (emsClient && lastLicenseId === currentLicenseId) {
+      return emsClient;
+    }
+
+    lastLicenseId = currentLicenseId;
+    if (emsSettings.isIncludeElasticMapsService()) {
```

Comment:
Should this include a check for lastLicenseId being undefined? What happens if  getLicenseId returns undefined?

Reply:
EMS has a license validation on the backend as well. This will fallback to as-if OSS EMS would be used.